sdk/search/Azure.Search/README.md

```diff
@@ -0,0 +1,171 @@
+mobile, and enterprise applications. Your code or a tool invokes data ingestion
+(indexing) to create and load an index. Optionally, you can add cognitive
+skills to apply AI processes during indexing. Doing so can add new information
+and structures useful for search and other scenarios.
+
+On the other side of your service, your application code issues query requests
+and handles responses.  The search experience is defined in your client using
+functionality from Azure Cognitive Search, with query execution over a
+persisted index that you create, own, and store in your service.
+
+[Source code][source] | [Package (NuGet)][package] | [API reference documentation][docs] | [REST API documentation][rest_docs] | [Product documentation][product_docs]
+
+## Getting started
+
+### Install the package
+
+Install the Azure Cognitive Search client library for .NET with [NuGet][nuget]:
+
+```Powershell
+dotnet add package Azure.Search --version 1.0.0-preview.1
+```
+
+### Prerequisites
+
+You need an [Azure subscription][azure_sub] and a
+[Search Service][create_search_service_docs] to use this package.
+
+To create a new Search Service, you can use the [Azure Portal][create_search_service_docs],
+[Azure PowerShell][create_search_service_ps], or the [Azure CLI][create_search_service_cli].
+Here's an example using the Azure CLI to create a free instance for getting started:
+
+```Powershell
+az search service create --name mysearch --resource-group mysearch-rg --sku free --location westus
+```
+
+## Key concepts
+
+Azure Cognitive Search is the only cloud search service with built-in AI
+capabilities that enrich all types of information to easily identify and
+explore relevant content at scale.
+
+- Fully managed search as a service to reduce complexity and scale easily
+- Auto-complete, geospatial search, filtering, and faceting capabilities for a rich user experience
+- Built-in AI capabilities including OCR, key phrase extraction, and named entity recognition to unlock insights
+- Flexible integration of custom models, classifiers, and rankers to fit your domain-specific needs
+
+`Azure.Search` offers two types of resources:
+
+- The _search service_ used via `SearchServiceClient` to create and manage search indexes
+- A _search index_ in the search service used via `SearchIndexClient` to query and manage the documents in a search index
+
+## Examples
+
+### Getting service statistics
+
+```C# Snippet:Azure_Search_Tests_Samples_GetStatisticsAsync
+// Create a new SearchServiceClient
+Uri endpoint = new Uri(Environment.GetEnvironmentVariable("SEARCH_ENDPOINT"));
+SearchApiKeyCredential credential = new SearchApiKeyCredential(
+    Environment.GetEnvironmentVariable("SEARCH_API_KEY"));
+SearchServiceClient search = new SearchServiceClient(endpoint, credential);
+
+// Get and report the Search Service statistics
+Response<SearchServiceStatistics> stats = await search.GetStatisticsAsync();
```

Comment:
No. All methods should return a `Response<T>`, so "WithResponse" would be redundant. `Response<T>` also has an implicit cast operator to `T` so people can use declared local variable types instead of `var` to just ignore the raw response if they want.

See https://azure.github.io/azure-sdk/general_introduction.html for more guidelines.

Reply:
Does the implicit conversion work in F#? We have customers using Search from F#. It would be unfortunate if the .NET guidelines lead to a sub-par experience in one of our supported .NET languages (I realize the guidelines are named "C#/.NET", but C# <> .NET).

Reply:
I don't know if the implicit cast operators work for F# (haven't used it enough to know/remember), but all they do is return the value of the public `Value` property which would be available to all .NET languages (CTS).

Reply:
Java does WithResponse.  .NET doesn't per the guidelines Heath linked.

F# does **not** expose the implicit cast, but I don't see `Response<T>.Value` as an undue burden for F# and other .NET languages.

Reply:
Fair enough.